### PR TITLE
post page added with functionality of adding a new post and delete an…

### PR DESCRIPTION
**Pull request by SibghatUllah-Devsinc**
… existing post

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// import axios from 'axios';
+
+// let instance = axios.config({
+//   URl: 'API'
+// })
+
```
Comment (zakria-devsinc) [inline src/api/index.js:6]: commented code

File: src/api/posts.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+
+// const getPost = () => axios.get('URL')
+
+// const addPost = (post) => axios.post('URL',post)
+
+// const updatePost = (post) => axios.put('URL',post)
+
+// const deletePost = (id) => axios.delete('URL')
```
Comment (zakria-devsinc) [inline src/api/posts.js:9]: commented code

File: src/components/Comments/CommentsList.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+//   }
+
+//   var data =await  response.json();
+//   if(localStorage.getItem("posts") == null){
+//     localStorage.setItem("posts", JSON.stringify(data));
+//     var localPosts = JSON.parse(localStorage.getItem('posts'));
+//     setLocalPost(localPosts)
+//     console.log(localPosts);
+//     }
+//     else{
```
Comment (zakria-devsinc) [inline src/components/Comments/CommentsList.js:18]: commented code

File: src/components/Comments/CommentsList.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+// getPosts().catch(error =>{
+//   setError(error.message)
+// });
+// } ,[])
+//   return (
+//     <ul className={classes.comments}>
+//       {props.comments.map((comment) => (
+//         <CommentItem key={comment.id} text={comment.comment} />
+//       ))}
+//     </ul>
```
Comment (zakria-devsinc) [inline src/components/Comments/CommentsList.js:32]: commented code

File: src/components/MainHeader/Navigation.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          <li>
+          <NavLink to="/addPost" activeClassName={classes.active}>Add New Post</NavLink>
+          </li>
+        )}
+        {ctx.isLoggedIn && (
+          <li>
+          <NavLink to="/deletePost" activeClassName={classes.active}>Delete Post Post</NavLink>
+          </li>
+        )}
+        {ctx.isLoggedIn && (
```
Comment (zakria-devsinc) [inline src/components/MainHeader/Navigation.jsx:27]: optimize code

File: src/components/Post/Delete.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import classes from './PostItem.module.css';
+
+
+
+
```
Comment (zakria-devsinc) [inline src/components/Post/Delete.jsx:5]: white spaces

File: src/components/Post/Delete.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+const DeleteItem = (props) => {
+const DeletePost = ()=>{
+  // const userPosts = props.userPost.filter((post)=> post.id !== props.id)
```
Comment (zakria-devsinc) [inline src/components/Post/Delete.jsx:8]: cmmented code

File: src/container/Login/Login.jsx [hunk 8]
```diff
@@ -90,10 +90,12 @@ const Login = (props) => {
...
+      const userId = existingUser.length + 1 
       existingUser.push({
         name:formState.nameValue,
         email:formState.emailValue,
-        password:formState.passwordValue
+        password:formState.passwordValue,
+        userId:"a"+ userId
       });
       localStorage.setItem('user', JSON.stringify(existingUser))
       history.push("/");
```
Comment (zakria-devsinc) [inline src/container/Login/Login.jsx:101]: useRouter use karo

File: src/container/Posts/AddPost.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+import {useHistory} from "react-router-dom"
+
+function AddPost(){
+    const history = useHistory()
+    function addPost(post){
+      var existingPosts = JSON.parse(localStorage.getItem('posts'));
+      existingPosts.push(post);
+      localStorage.setItem('posts', JSON.stringify(existingPosts))
+      history.push("/home");
+      console.log("added")
```
Comment (zakria-devsinc) [inline src/container/Posts/AddPost.js:11]: remove consoles

File: src/container/Posts/AddPost.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import PostForm from "./PostForm"
+import {useHistory} from "react-router-dom"
+
+function AddPost(){
+    const history = useHistory()
+    function addPost(post){
+      var existingPosts = JSON.parse(localStorage.getItem('posts'));
```
Comment (zakria-devsinc) [inline src/container/Posts/AddPost.js:7]: use const

File: src/container/Posts/AllPosts.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+import NoPostsFound from "../../components/Post/NoPostsFound";
+import PostList from "./PostList"
+
+
```
Comment (zakria-devsinc) [inline src/container/Posts/AllPosts.js:5]: spaces

File: src/container/Posts/AllPosts.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+function AllPosts(){
+
```
Comment (zakria-devsinc) [inline src/container/Posts/AllPosts.js:7]: space

File: src/container/Posts/AllPosts.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+const [error , setError] = useState()
+const [localpost , setLocalPost] = useState()
+  useEffect( ()=>{
+
```
Comment (zakria-devsinc) [inline src/container/Posts/AllPosts.js:11]: space

File: src/container/Posts/AllPosts.js [hunk 14]
```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+import NoPostsFound from "../../components/Post/NoPostsFound";
+import PostList from "./PostList"
+
+
+function AllPosts(){
+
+const [error , setError] = useState()
+const [localpost , setLocalPost] = useState()
```
Comment (zakria-devsinc) [inline src/container/Posts/AllPosts.js:9]: handler or state ka darmyan space enter kia karo

File: src/container/Posts/AllPosts.js [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+  const getPosts= async() =>{
+  const response = await fetch('https://jsonplaceholder.typicode.com/posts');
+  if (!response.ok) {
+    throw new Error('Something went wrong! '+ response.status);
+  }
+
+  const data =await  response.json();
+  if(localStorage.getItem("posts") == null){
+    localStorage.setItem("posts", JSON.stringify(data));
+    var localPosts = JSON.parse(localStorage.getItem('posts'));
```
Comment (zakria-devsinc) [inline src/container/Posts/AllPosts.js:21]: yar ye var khatam karo saray

File: src/container/Posts/AllPosts.js [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    setLocalPost(localPosts)
+    console.log(localPosts);
```
Comment (zakria-devsinc) [inline src/container/Posts/AllPosts.js:23]: consoles b saray remove karo

File: src/container/Posts/DeletePost.js [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+import NoPostsFound from '../../components/Post/NoPostsFound';
+import newContext from '../../components/Context/Auth-Context';
+import {useHistory} from "react-router-dom"
+
+const DeletePost = () => {
+  const ctx = useContext(newContext)
+  const history = useHistory()
+  const [localpost , setLocalPost] = useState()
+  const [allPosts , setAllPosts] = useState()
+  console.log(ctx.userId);
```
Comment (zakria-devsinc) [inline src/container/Posts/DeletePost.js:13]: console

File: src/utils/utils.js [hunk 18]
```diff
@@ -0,0 +1,3 @@
+// const getToken = () => {
+//   //
+// }
```
Comment (zakria-devsinc) [inline src/utils/utils.js:3]: remove comments

File: src/container/Posts/PostList.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
+import { Fragment } from 'react';
+import PostItem from '../../components/Post/PostItem';
+import classes from './PostList.module.css';
+
+const PostList = (props) => {
+  
+ var posts = props.allPosts
```
Comment (zakria-devsinc) [inline src/container/Posts/PostList.js:7]: use const

File: src/components/Post/NoPostsFound.js [hunk 20]
```diff
@@ -0,0 +1,15 @@
+import classes from './NoPostsFound.module.css';
+import {Link} from "react-router-dom"
+
+const NoPostsFound = () => {
```
Comment (zakria-devsinc) [inline src/components/Post/NoPostsFound.js:4]: line break

File: src/container/Posts/PostForm.js [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+  const [isEntered, setIsEntered]=useState(false)
+  const titleInputRef = useRef();
+  const textInputRef = useRef();
+  function submitFormHandler(event) {
+    event.preventDefault();
+
+    const enteredAuthor = titleInputRef.current.value;
+    const enteredText = textInputRef.current.value;
+    const localPosts = JSON.parse(localStorage.getItem('posts'));
+    console.log(localPosts.length)
```
Comment (zakria-devsinc) [inline src/container/Posts/PostForm.js:18]: consoles

**Review by zakria-devsinc — CHANGES_REQUESTED** [review]

**Comment by SibghatUllah-Devsinc** [comment]
updation done

**Pull request merged**